File: ios/VllowDugout/Models/PerformanceStats.swift
import Foundation

nonisolated struct PerformanceStats: Sendable {
    let runsScored: Int
    let ballsFaced: Int
    let strikeRateChange: Double
    let opponent: String
    let coinBonus: Int

    var strikeRate: Double {
        guard ballsFaced > 0 else { return 0 }
        return (Double(runsScored) / Double(ballsFaced)) * 100
    }
}

nonisolated struct ScoringZone: Identifiable, Sendable {
    let id: UUID
    let name: String
    let runs: Int
    let xPosition: Double
    let yPosition: Double
    let intensity: Double

    init(id: UUID = UUID(), name: String, runs: Int, xPosition: Double, yPosition: Double, intensity: Double) {
        self.id = id
        self.name = name
        self.runs = runs
        self.xPosition = xPosition
        self.yPosition = yPosition
        self.intensity = intensity
    }
}

nonisolated struct DismissalPattern: Identifiable, Sendable {
    let id: UUID
    let type: String
    let percentage: Double

    init(id: UUID = UUID(), type: String, percentage: Double) {
        self.id = id
        self.type = type
        self.percentage = percentage
    }
}

File: flutter_app/lib/models/performance_stats.dart
import 'package:uuid/uuid.dart';

class PerformanceStats {
  final int runsScored;
  final int ballsFaced;
  final double strikeRateChange;
  final String opponent;
  final int coinBonus;

  PerformanceStats({
    required this.runsScored,
    required this.ballsFaced,
    required this.strikeRateChange,
    required this.opponent,
    required this.coinBonus,
  });

  double get strikeRate {
    if (ballsFaced <= 0) return 0;
    return (runsScored / ballsFaced) * 100;
  }
}

class ScoringZone {
  final String id;
  final String name;
  final int runs;
  final double xPosition;
  final double yPosition;
  final double intensity;

  ScoringZone({
    String? id,
    required this.name,
    required this.runs,
    required this.xPosition,
    required this.yPosition,
    required this.intensity,
  }) : id = id ?? const Uuid().v4();
}

class DismissalPattern {
  final String id;
  final String type;
  final double percentage;

  DismissalPattern({
    String? id,
    required this.type,
    required this.percentage,
  }) : id = id ?? const Uuid().v4();
}
